web/src/components/Layout/LoggedUserDisplay/hooks/user-hooks.ts

```diff
@@ -1,58 +1,77 @@
 import { toast } from "react-toastify";
 
 import {
-  useFetchUserQuery,
+  useFetchUserLazyQuery,
   FetchUserDocument,
 } from "../graphql/fetch-user.generated";
 
 import { useUpdateUserMutation } from "../graphql/update-user.generated";
 import { useRouter } from "next/router";
-import { UserContext, useUser } from "@auth0/nextjs-auth0/client";
+import { useUser } from "@auth0/nextjs-auth0/client";
+import { TeamsDocument } from "@/scenes/team/graphql/teams.generated";
+import { Auth0SessionUser, Auth0User } from "@/lib/types";
+import { urls } from "@/lib/urls";
+import { useEffect, useMemo } from "react";
 
 export const useFetchUser = () => {
-  const { user: auth0User, ...rest } = useUser() as Omit<
-    UserContext,
-    "user"
-  > & {
-    user: UserContext["user"] & {
-      hasura: {
-        id: string;
-        auth0Id: string;
-        team_id: string;
-      };
-    };
-  };
-
   const router = useRouter();
+  const team_id = router.query.team_id as string | undefined;
+  const { user: auth0User, ...rest } = useUser() as Auth0SessionUser;
 
-  const { data, ...other } = useFetchUserQuery({
-    variables: { id: auth0User?.hasura?.id },
+  const [fetchUser, { data, ...other }] = useFetchUserLazyQuery({
+    context: { headers: { team_id } },
+    variables: { id: auth0User?.hasura?.id ?? "" },
 
     onCompleted: (data) => {
       if (!data.user[0]) {
-        router.push("/logout");
+        toast.error("Something went wrong. Please try again.");
```

Comment:
how about adding a number error code here so the users can reference if reaching out for support

Reply:
made it a 404 error